build/source/driver/multi_driver.f90

```diff
@@ -233,23 +234,87 @@ program multi_driver
 ! output control 
 integer(i4b)                     :: iFreq                      ! index for looping through output files
 
+! parallelize the model run
+integer(i4b)                     :: startGRU                   ! index of the starting GRU for parallelization run
+integer(i4b)                     :: checkHRU                   ! index of the HRU for a single HRU run
+character(len=256),allocatable   :: argString(:)               ! string to store command line arguments
+integer(i4b)                     :: nArgument                  ! number of command line arguments
+integer(i4b)                     :: skipArgument               ! number of command line arguments to skip
+integer(i4b)                     :: iArgument                  ! index of command line argument
+integer(i4b)                     :: maxGRU                     ! maximum number of GRUs in the input file
+character(len=128)               :: fmtGruOutput               ! a format string used to write start and end GRU in output file names
 ! *****************************************************************************
 ! (1) inital priming -- get command line arguments, identify files, etc.
 ! *****************************************************************************
-print*, 'start'
 ! get the initial time
-call date_and_time(cdate1,ctime1)
-print*,ctime1
-! get command-line arguments for the output file suffix
-call getarg(1,output_fileSuffix)
-if (len_trim(output_fileSuffix) == 0) then
- print*,'1st command-line argument missing, expect text string defining the output file suffix'; stop
-endif
-! get command-line argument for the muster file
-call getarg(2,summaFileManagerFile) ! path/name of file defining directories and files
-if (len_trim(summaFileManagerFile) == 0) then
- print*,'2nd command-line argument missing, expect path/name of muster file'; stop
-endif
+call date_and_time(values=ctime1)
+print "(A,I2.2,':',I2.2,':',I2.2)", 'start at ',ctime1(5:7)
+
+! check numbers of command-line arguments and obtain all arguments 
+nArgument = getCommandArguments(argString)
+
+if (nArgument < 1) then 
+ call printCommandHelp()
+end if
+
+! initialize command line argument variables
+startGRU=integerMissing; nGRU=integerMissing; checkHRU=integerMissing; nHRU=integerMissing
+
+! loop through all command arguments
+skipArgument = 0
```

Comment:
Use `nLocalArguments` or something similae -- since we are not actually skipping arguments